File: 6-Data_structures_arrays/lists.py
strings = ["a", "b", "c", "d"]

# lookup
print(strings[2])  # => O(n)

# append
strings.append("e")  # => O(n)

# pop
strings.pop()  # => O(n)

# insert
strings.insert(0, "x")  # => O(n)
strings.insert(2, "alien")

print(strings)

File: 6-Data_structures_arrays/lists/lib/lists.ex
defmodule Lists do
  def list(lst) do
    # lookup
    # => O(n)
    Enum.at(lst, 2)

    # appending
    # => O(n)
    _ = lst ++ ['e']

    # delete_at
    # => O(n)
    List.delete_at(lst, -1)

    # prepending
    # => O(1)
    ['x' | lst]

    # insert_at
    # => O(n)
    List.insert_at(lst, 2, 'alien')
  end
end

# Lists.list(["a", "b", "c", "d"])
